code/clue.js

```diff
@@ -1,59 +1,460 @@
+  name: 'Spa',
+  image: 'assets/spa.png',
+}
+
+const livingRoom = {
+  name: 'Living Room',
+  image: 'assets/living-room.png',
+}
+
+const observatory = {
+  name: 'Observatory',
+  image: 'assets/observatory.png',
+}
+
+const theater = {
+  name: 'Theater',
+  image: 'assets/theater.png',
+}
+
+const guestHouse = {
+  name: 'Guest House',
+  image: 'assets/guest-house.png',
 }
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const patio = {
+  name: 'Patio',
+  image: 'assets/patio.png',
+}
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+// Arrays
 
 const suspects = [
   mrGreen,
+  profPlum,
+  msScarlet,
+  mrsPeacock,
+  colMustard,
   mrsWhite
-  // ...  and the rest
 ]
 
-const weapons = []
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+]
+
+const rooms = [
+  diningRoom,
+  conservatory,
+  kitchen,
+  study,
+  library,
+  billiardRoom,
+  lounge,
+  ballroom,
+  hall,
+  spa,
+  livingRoom,
+  observatory,
+  theater,
+  guestHouse,
+  patio
+]
+
+// Variables
 
-const rooms = []
+const killerCard = document.getElementById('killerCard');
+const killerName = document.getElementById('killerName');
+const killerAge = document.getElementById('killerAge');
+const killerImage = document.getElementById('killerImage');
+const killerFavouriteWeapon = document.getElementById('killerFavouriteWeapon');
+const weaponCard = document.getElementById('weaponCard');
+const weaponName = document.getElementById('weaponName');
+const weaponWeight = document.getElementById('weaponWeight');
+const weaponImage = document.getElementById('weaponImage');
+const weaponFavouriteRoom = document.getElementById('weaponFavouriteRoom');
+const roomCard = document.getElementById('roomCard');
+const roomName = document.getElementById('roomName');
+const roomImage = document.getElementById('roomImage');
+const revealedMystery = document.getElementById('mystery');
+const revealButton = document.getElementById('revealButton');
+const resetButton = document.getElementById('resetButton');
+const shuffleButton = document.getElementById('shuffleButton');
+const killerLoader = document.getElementById('killerLoader');
+const weaponLoader = document.getElementById('weaponLoader');
+const roomLoader = document.getElementById('roomLoader');
+
+// Shuffle function
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// Mystery storage
+
+const mystery = {
+  killer: undefined,
+  favouriteWeapon: undefined,
+  pickedWeapon: undefined,
+  weapon: undefined,
+  favouriteRoom: undefined,
+  pickedRoom: undefined,
+  room: undefined,
```

Comment:
It's even more semantic approach to initialise mystery object with null values like this :

```
const mystery = {
  killer: null,
  favouriteWeapon: null,
  pickedWeapon: null,
  weapon: null,
  favouriteRoom: null,
  pickedRoom: null,
  room: null,
};
```